cmd/explaintest/r/topn_pushdown.result

```diff
@@ -0,0 +1,9 @@
+explain select * from ((select 4 as a) union all (select 33 as a)) tmp order by a desc limit 1;
+id	count	task	operator info
+TopN_17	1.00	root	tmp.a:desc, offset:0, count:1
+└─Union_21	2.00	root	
+  ├─Projection_22	1.00	root	cast(a)
+  │ └─Projection_23	1.00	root	4
```

Comment:
If we have a rule to merge sequential `Project` operators, these two `Project` operator can be merged to one.

Reply:
If we change `33` to `3`, would this `Projection_22` be removed?

Reply:
Yes, because in this time, FieldTypes of "3" and "4" are equal, no cast function will be added in the logical plan building phase.